chainer/variable.py

```diff
@@ -313,6 +313,16 @@ def _set_grad_var_if_available(self, g):
         if var is not None:
             var._set_grad_var_without_check(g)
 
+    def _pop_grad_var_if_available(self):
```

Comment:
How about making this method a member of `GradTable`?
This set of operations does not seem to make sense for `VariableNode`.

(You can simply use `VariableNode.variable_or_none()` and `Variable.grad_var.setter`.)